GPU/Common/FragmentShaderGenerator.cpp

```diff
@@ -819,10 +814,9 @@ bool GenerateFragmentShader(const FShaderID &id, char *buffer, const ShaderLangu
 						WRITE(p, "  vec3 colortest = roundAndScaleTo255v(v.rgb);\n");
 						WRITE(p, "  if ((colortest.r %s u_alphacolorref.r) && (colortest.g %s u_alphacolorref.g) && (colortest.b %s u_alphacolorref.b)) %s\n", test, test, test, discardStatement);
 					} else if (compat.bitwiseOps) {
-						// Apparently GLES3 does not support vector bitwise ops.
 						WRITE(p, "  ivec3 v_scaled = roundAndScaleTo255iv(v.rgb);\n");
 						if (compat.shaderLanguage == GLSL_VULKAN) {
-							// TODO: Use this for GL as well?
+							// Apparently GLES3 does not support vector bitwise ops, but Vulkan does?
```

Comment:
Oh, right, it's not just about bitwise but vector bitwise which was weirdly left out of ES3?

Reply:
Correct, ivec3 & ivec3 is not supported, btu int & int is.  So we have to do it by component outside Vulkan.

-[Unknown]